src/AspNet.Security.OAuth.Nextcloud/NextcloudAuthenticationHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Nextcloud
+{
+    public class NextcloudAuthenticationHandler : OAuthHandler<NextcloudAuthenticationOptions>
+    {
+        public NextcloudAuthenticationHandler(
+            [NotNull] IOptionsMonitor<NextcloudAuthenticationOptions> options,
+            [NotNull] ILoggerFactory logger,
+            [NotNull] UrlEncoder encoder,
+            [NotNull] ISystemClock clock)
+            : base(options, logger, encoder, clock)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
+        {
+            string userId = tokens.Response.Value<string>("user_id");
+            string userEndpoint = Options.UserInformationEndpoint.TrimEnd('/');
+            userEndpoint += $"/{userId}";
```

Comment:
`userEndpoint += '/' + Uri.EscapeUriString(userId);`

or 

`userEndpoint += '/' + Uri.EscapeDataString(userId);`

(I can't remember which is the right one)

Otherwise if the username contains any special characters it will be malformed.

Reply:
```
            userEndpoint += $"/{Uri.EscapeUriString(userId)}";
```
Should be ok?

Btw currently: Only the following characters are allowed in a username: "a-z", "A-Z", "0-9", and "_.@-'"

Reply:
I think that's right - maybe could change the user name in the test case to include an `@` to verify it encodes correctly?